### ISSUE TITLE
Adding a 'Watch' view to the debugging facilities

### ISSUE THREAD

**Issue by TobyLobster** [tom-seddon/b2#130]
When debugging, it is often useful to keep track of the values of certain memory locations as the code is stepped through. So this request is for a new view (a 'Watch' view) with the ability to add requested specific memory locations, optionally giving each a descriptive string name. The values would get updated as the code is stepped through. If these watch variables could remain even between different launches of the application, that would be useful. Stretch goals: allow showing either a single 8 bit byte or a 16 bit value from a pair of consecutive bytes. Show values as both decimal, hex, and ASCII.

**Comment by tom-seddon** [comment]
I'd been holding off on adding watch windows - it's felt like it'd need a proper expression evaluator and all that. But I think you're right, and this suggestion would suffice for most cases.

Multiple data types is a must! Would be nice to be able to see a few different types of string, at least 0- and CR-terminated. Might be nice to visualize bytes as pixels too.

**Comment by TobyLobster** [comment]
Yes, I could see a version 2 or 3 of this solution having fancy features like 'break when this value goes above 127' and such, but a quite basic version 1 would help a lot of the problem case I think.

**Comment by tom-seddon** [comment]
This functionality could potentially cover #172 and #173. Any values you want to copy, display in the watch window, then copy them out of the value column.

**Comment by TheOriginalSR** [comment]
I know this is an old request however, I have just had this exact situation arise that it would be useful to add a Byte/Word/Double Word Watch to a memory location - especially if you could label that location/watch too.

Break on expression is an absolute luxury.

My only workaround so far is a second disassembly debug window with a couple of post-its to help me remember!
